lib/CPython/Simple/Instances.hs

```diff
@@ -0,0 +1,96 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module CPython.Simple.Instances where
+
+import Control.Monad ((<=<))
+import Data.Text (Text)
+import qualified Data.Text as T
+
+import qualified CPython.Constants as Py
+import qualified CPython.Protocols.Object as Py
+import qualified CPython.Types as Py
+import qualified CPython.Types.Tuple as Py (fromTuple)
+
+-- TODO: ToPy/FromPy for Bool will require some library changes (e.g. adding fromBool)
+
+class ToPy a where
+  toPy :: a -> IO Py.SomeObject
+
+class FromPy a where
+  fromPy :: Py.SomeObject -> IO a
+
+easyToPy
+  :: Py.Object c
+  => (a -> IO c) -- ^ python to- conversion, e.g. Py.toFloat
+  -> a           -- ^ haskell type being converted
+  -> IO Py.SomeObject
+easyToPy f = fmap Py.toObject . f
+
+easyFromPy
+  :: Py.Concrete b
+  => (b -> IO c)   -- ^ python from- conversion, e.g. Py.fromFloat
+  -> Text          -- ^ error message text for the haskell type being converted to, e.g. "Double"
+  -> Py.SomeObject -- ^ python object to cast from
+  -> IO c
+easyFromPy conversion typename = conversion . castErr typename <=< Py.cast
+  where castErr typename' = \case
+          Nothing -> error $ "FromPy could not cast to " ++ T.unpack typename'
```

Comment:
this could definitely worth its own exception to be created, specific to this module, like `PyConversionError` or something